src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -619,17 +634,77 @@ ServerHandlerImp::processRequest (Port const& port,
 // Returns `true` if the HTTP request is a Websockets Upgrade
 // http://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header#Use_with_WebSockets
 bool
-ServerHandlerImp::isWebsocketUpgrade (http_request_type const& request)
+ServerHandlerImp::isWebsocketUpgrade(
+    http_request_type const& request) const
 {
     if (is_upgrade(request))
-        return request.headers["Upgrade"] == "websocket";
+        return beast::detail::ci_equal(
+            request.headers["Upgrade"], "websocket");
     return false;
 }
 
+bool
+ServerHandlerImp::isStatusRequest(
+    http_request_type const& request) const
+{
+    return request.version >= 11 && request.url == "/" &&
+        request.body.size() == 0 && request.method == "GET";
+}
+
+Handoff
+ServerHandlerImp::statusRequest(
+    http_request_type const& request) const
+{
+    using namespace beast::http;
+    Handoff handoff;
+    response_v1<string_body> msg;
+    std::string reason;
+    if (app_.serverOkay(reason))
+    {
+        msg.status = 200;
+        msg.reason = "OK";
+        msg.body = "<!DOCTYPE html><html><head><title>" + systemName () +
+            " Test page for rippled</title></head><body><h1>" +
+                systemName () + " Test</h1><p>This page shows rippled http(s) "
+                    "connectivity is working.</p></body></html>";
+    }
+    else
+    {
+        msg.status = 500;
+        msg.reason = "Internal Server Error";
+        msg.body = "<HTML><BODY>Server cannot accept clients: " +
+            reason + "</BODY></HTML>";
+    }
+    msg.version = request.version;
+    msg.headers.insert("Server", BuildInfo::getFullVersionString());
+    msg.headers.insert("Content-Type", "text/html");
+    prepare(msg, beast::http::connection::close);
+    handoff.response = std::make_shared<SimpleWriter>(msg);
+    return handoff;
+}
+
+Handoff
+ServerHandlerImp::unathorizedResponse(
+    http_request_type const& request) const
+{
+    using namespace beast::http;
+    Handoff handoff;
+    response_v1<string_body> msg;
+    msg.version = request.version;
+    msg.status = 401;
+    msg.reason = "Unauthorized";
+    msg.headers.insert("Server", BuildInfo::getFullVersionString());
+    msg.headers.insert("Content-Type", "text/html");
+    msg.body = "Invalid protocol.";
+    prepare(msg, beast::http::connection::close);
+    handoff.response = std::make_shared<SimpleWriter>(msg);
+    return handoff;
+}
+
 // VFALCO TODO Rewrite to use beast::http::headers
 bool
 ServerHandlerImp::authorized (Port const& port,
-    std::map<std::string, std::string> const& h)
+    std::map<std::string, std::string> const& h) const
```

Comment:
`const` is somewhat frivolous here because we never defined what operations are valid on a `ServerHandlerImp const`. And to my knowledge, no server code invokes handlers through a `const` this pointer. Its certainly not hurting, but it also doesn't quite mean what you think it means.